plugins/auth-backend/src/providers/microsoft/provider.ts

```diff
@@ -124,48 +143,57 @@ export class MicrosoftAuthProvider implements OAuthHandlers {
         req.scope,
       );
 
-    const fullProfile = await executeFetchUserProfileStrategy(
-      this._strategy,
-      accessToken,
-    );
-
     return {
       response: await this.handleResult({
-        fullProfile,
         params,
         accessToken,
+        ...(!this.skipUserProfile(accessToken) && {
+          fullProfile: await executeFetchUserProfileStrategy(
+            this._strategy,
+            accessToken,
+          ),
+        }),
       }),
       refreshToken,
```

Comment:
To double check this: we'll be issuing a lot of different tokens, but they're all stored in the same refresh token cookie. Is that fine? Does this refresh token refer to a session on the MS side that has the union of all scopes we've been requesting, even if they are for different resources?

Basically, if I first sign-in with MS auth, and then head over to the k8s cluster page and request access to those resources, I'd expect one auth popup for each of those steps. If I do the same thing again, I'd expect both of those authorizations to be part of the session already and neither of those two actions should trigger an auth popup again.

Reply:
The answer to these question is a resounding 'Yes' -- Grants like authorization codes or refresh tokens are NOT subject to the same "only one resource" constraint that access tokens have. You can use a refresh token to get an access token for any consented scope from any resource, regardless of how you got that refresh token.

Hopefully the language in c15e0cedbe1981ccbe29bba83524d30b5357f74f makes this more clear:

> gets access token for other consented scopes besides those directly granted

Because this is really important -- it's the only reason any of this is viable!

Reply:
Excellent! 😁